src/server_mocker/tcp_server_mocker.rs

```diff
@@ -27,7 +27,7 @@ use std::time::Duration;
 pub struct TcpServerMocker {
     socket_addr: SocketAddr,
     instruction_tx: Sender<Vec<Instruction>>,
-    message_rx: Receiver<BinaryMessage>,
+    message_rx: Receiver<Vec<u8>>,
```

Comment:
Why removing this type? You prefer `Vec<u8>` over `BinaryMessage`?

This is a practice I saw on some other crates.

Reply:
Type aliasing in Rust mostly obfuscates the meaning, without that much value: Clippy for example has failed to notice some optimizations, and users will not realize this is a vector and how to deal with it.

The only common case of type aliasing I have seen is for `Result`, e.g. `MockResult<T> = Result<T, MockError>` -- this one is pretty common.

Reply:
P.S. rarely it does make sense to add type aliasing - but that's usually for giant types like `Box<Option<Foo<Bar>>>>`, and even then it might be better to create a simple wrapper that implements `Deref` trait.  See [docs](https://doc.rust-lang.org/std/ops/trait.Deref.html#when-to-implement-deref-or-derefmut)